### PR TITLE
Simplify Tinflex API

### PR DESCRIPTION
**Pull request by wilzbach**
I think it's a reasonable assumption that all functions have the same type?

(see also #240)

### CODE REVIEW

**Comment by 9il** [comment]
please keep old api

**Comment by wilzbach** [comment]
> please keep old api

ok, but you explicitly instructed me on #240 for this API and it made sense to me as the pdf is used for sampling.

**Comment by wilzbach** [comment]
> please keep old api

rebased.

File: source/mir/random/flex/package.d [hunk 1]
```diff
@@ -110,6 +111,10 @@ Returns:
 auto flex(F0, F1, F2, S)
```
Comment (9il) [inline source/mir/random/flex/package.d]: `S` should be the first argument.

File: source/mir/random/flex/package.d [hunk 2]
```diff
@@ -110,6 +111,10 @@ Returns:
...
                (in F0 f0, in F1 f1, in F2 f2,
                 S c, S[] points, S rho = 1.1)
+    if (isFloatingPoint!(ReturnType!F0) &&
```
Comment (9il) [inline source/mir/random/flex/package.d]: use `allSatisfy`

Comment (9il) [inline source/mir/random/flex/package.d]: BTW, function can return integer

File: source/mir/random/flex/package.d [hunk 3]
```diff
@@ -118,18 +123,52 @@ auto flex(F0, F1, F2, S)
 }
 
 /// ditto
+auto flex(Pdf, F0, F1, F2, S)
```
Comment (9il) [inline source/mir/random/flex/package.d]: Ditto

File: source/mir/random/flex/package.d (hunk 2 above)
Comment (9il) [inline source/mir/random/flex/package.d]: Just remove this constraints. They would not work for templated opCall for example

File: source/mir/random/flex/package.d [hunk 4]
```diff
@@ -118,18 +123,52 @@ auto flex(F0, F1, F2, S)
...
+               (in Pdf pdf, in F0 f0, in F1 f1, in F2 f2,
+                S c, S[] points, S rho = 1.1)
+    if (isFloatingPoint!(ReturnType!Pdf) &&
```
Comment (9il) [inline source/mir/random/flex/package.d]: remove this constraints. Save only `isFloatingPoint!S`

**Comment by wilzbach** [comment]
rebased API

File: source/mir/random/flex/package.d [hunk 5]
```diff
@@ -118,26 +120,49 @@ auto flex(F0, F1, F2, S)
...
 }
 
 /// ditto
-auto flex(F0, S)(in F0 f0, in FlexInterval!S[] intervals)
+auto flex(S, F0, F1, F2)
+               (in F0 f0, in F1 f1, in F2 f2,
+                S[] cs, S[] points, S rho = 1.1)
+    if (isFloatingPoint!S)
 {
     import std.math: exp;
```
Comment (9il) [inline source/mir/random/flex/package.d:140]: use internal math

**Pull request merged**